doc/dune

```diff
@@ -1,5 +1,7 @@
 (rule
  (targets unreleased.rst)
+ ; We need to keep this as the sphinx build still requires this file
+ ; in-tree, to be fixed hopefully soon
```

Comment:
Actually, I think this is incorrect because when building the refman through Dune, the build is already done out of tree (`Makefile.doc` is not used in that case).

Reply:
This is needed for the case you do a regular make build, then a dune build, otherwise dune will complain about the generated file; once we more the sphinx out of tree, this can be removed.

Reply:
I see, thanks!